advanced/pihole-admin.conf

```
@@ -15,13 +15,22 @@
 $HTTP["url"] =~ "^/admin/" {
     server.document-root = "/var/www/html"
     server.stream-response-body = 1
+
     fastcgi.server = (
         ".php" => (
             "localhost" => (
-                "socket" => "/tmp/pihole-php-fastcgi.socket",
+                "socket" => "/run/lighttpd/php.socket",
```

I think we can safely leave this as `/tmp/pi-hole-php-fastcgi.socket` - less chance of collision, then.

```suggestion
                "socket" => "/tmp/pihole-php-fastcgi.socket",
```

---

Depends on the user running the daemons and the access for the directories.

---

How do you mean? As to whether or not it would cause conflicts? Unless you can think of a reason why not, I feel like the socket being in /tmp for the pi-hole specific fastcgi.server config makes sense?

---

A couple of reasons offhand.

* Do we know that the daemon can write to `/tmp`? I honestly don't know if it can or can not.
* Do we know that `/tmp` is safe to write to, security wise? Can anyone see or access that socket and what are the implications of that.
* Most pressing, do we know that `/tmp` won't get purged by the system since it's really just meant for temporary storage. What happens if `/tmp/pihole-php-fastcgi.socket` is deleted while `lighttpd` is still running?

---

rd mentioned this in the PR message as well.

>Also, the new socket path was causing a permission error when the debug log was executed from the web interface.
This PR fixes the config.

---

Just looking over the `lighttpd` docs, all the examples are in `/tmp`, is all.

https://redmine.lighttpd.net/projects/lighttpd/wiki/Mod_fastcgi#:~:text=Starting%20a%20FastCGI%2DPHP

---

I'm really undecided here.

I agree with the idea of using a specific path. I searched the `lighttpd` docs and found only 2 or 3 examples using `/run`... the rest always uses `/tmp`.

On the other hand, I'm not sure about permissions on this path on each distro.
Also, I downloaded the packages for Debian 10 and 11. The original value used on `15-fastcgi-php.conf` file is always using `/run/lighttpd/` _(the one we always used)_.
On Fedora, the default path is initially defined as `/var/lib/lighttpd/sockets/`.

---

After considering the whole discussion here, a lot of testing and reviewing the previous settings, I decided to use the default `15-fastcgi-php.conf` (installed using `lighty-enable-mod fastcgi-php`) and remove the `fastcgi.server` option from the new file.

I will change the PR files.